sql/core/src/test/resources/sql-tests/inputs/time.sql

```diff
@@ -18,3 +18,35 @@ select try_to_time('12:48:31 abc');
 select try_to_time('10:11:12.', 'HH:mm:ss.SSSSSS');
 select try_to_time("02-69", "HH-mm");
 select try_to_time('11:12:13', 'HH:mm:ss', 'SSSSSS');
+
+select secondoftime_with_fraction(to_time('23-59-59.987654', 'HH-mm-ss.SSSSSS'));
+select second(to_time('23-59-58.987654', 'HH-mm-ss.SSSSSS'));
+select minute(to_time('23-59-58.987654', 'HH-mm-ss.SSSSSS'));
+select hour(to_time('23-59-58.987654', 'HH-mm-ss.SSSSSS'));
+
+select extract(HOUR from  to_time('23-59-58.987654', 'HH-mm-ss.SSSSSS'));
+select extract(H from  to_time('23-59-58.987654', 'HH-mm-ss.SSSSSS'));
+select extract(HOURS from  to_time('23-59-58.987654', 'HH-mm-ss.SSSSSS'));
+select extract(HR from  to_time('23-59-58.987654', 'HH-mm-ss.SSSSSS'));
+select extract(HRS from  to_time('23-59-58.987654', 'HH-mm-ss.SSSSSS'));
```

Comment:
As you are testing different `field` names, could you check different hours, for example:
```sql
select extract(HOUR from  time'00:59:00.987654');
select extract(H from  time'23:59:00.987654');
```
especially check some corner cases like `00:00:00` and `23:59:59.999999`

Reply:
Done